pkg/binlog/common/replication.go

```diff
@@ -32,6 +32,9 @@ func SetDefaultReplicationCfg(cfg *replication.BinlogSyncerConfig, retryCount in
 	cfg.UseDecimal = true // must set true. ref: https://github.com/pingcap/tidb-enterprise-tools/pull/272
 	cfg.VerifyChecksum = true
 	cfg.MaxReconnectAttempts = retryCount
+	if retryCount <= 1 {
```

Comment:
NOTE: in go-mysql, `zero or negative number means infinite retry. this configuration will not work if DisableRetrySync is true`.

Reply:
add comment in https://github.com/pingcap/dm/pull/655/commits/c352270d08dadf560cc5b2c32a6664bcc03d4d6f